spec/integration/driver_spec.rb

```diff
@@ -439,6 +439,84 @@ def create_screenshot(file, *args)
         @session.visit('/poltergeist/redirect_to_headers')
         expect(@driver.body).not_to include('X_CUSTOM_HEADER: 1')
       end
+
+      context 'multiple windows' do
+        it 'persists headers across popup windows' do
+          @driver.headers = {
+            'Cookie' => 'foo=bar',
+            'Host' => 'foo.com',
+            'User-Agent' => 'foo'
+          }
+          @session.visit('/poltergeist/popup_headers')
+          @session.click_link 'pop up'
+          @session.switch_to_window @session.windows.last
```

Comment:
`@session.windows` doesn't technically guarantee order - use `@session.window_opened_by` on previous line to capture the window object of the new object rather than assuming last